### ISSUE TITLE
Support for sample name while visualizing multisample vcf

### ISSUE THREAD

**Issue by man4ish** [igvteam/igv.js#1092]
Hi,

I am trying to visualize multisample vcf and it shows the sample name on mouseover but not on left hand side like desktop version.

Output from Igv.js
![1000Genome_Chr1](https://user-images.githubusercontent.com/6385968/79075561-9e06ce80-7cc1-11ea-9587-7a07f4d668c8.png)

<br><br><br><br>

Screen shot from youtube video for IGV (Desktop Version):
![Screenshot from 2020-04-12 13-28-01](https://user-images.githubusercontent.com/6385968/79075587-cabae600-7cc1-11ea-9883-8aba67a1f702.png)

How can i add sample name to left side?

Thanks

**Comment by jrobinso** [comment]
No you can't, currently this is not a feature of the web version.  It will
come but is not there yet.

**Comment by man4ish** [comment]
Thanks

**Comment by jrobinso** [comment]
@man4ish I going to reopen this because I don't think we have an open issue for this feature, which is an important one for VCF and also copy number.

**Comment by man4ish** [comment]
can you pls add that feature in coming week as i need it for my project. I will also site your code link in my publication.

**Comment by jrobinso** [comment]
@man4ish There is no possibility of that happening.

**Comment by amanpatel** [comment]
We need this feature too. Can I get any developer related pointers to this? I can try to implement this in the next week.

I've written my own customer reader and source classes for a VCF track that is actually "powered" by the database in a python middleware.  I have never worked with canvas, but this seems like it could be done via svg+css.

**Comment by jrobinso** [comment]
@amanpatel  This is on our list for the next round of work, which will begin in a few weeks.   Thanks for the offer but I do not encourage a pull request for this, it touches too many fundamental classes and DOM organization.

**Comment by amanpatel** [comment]
I understand this is in the next sprint. Any foresight to when this could be realized.

In the mean time, I'm looking for a hack around to achieve this. I'm thinkin I can create a floating (non-movable) window that has the sample names spread out vertically that corresponds to the exact track position. But obviously determining the height of each sample's variants will be tricky (as they sometimes take more than 1 track for the same sample).

If that works, I can just attach some event listeners to the div to have it place in the appropriate area on the left side of the IGV browser.

**Comment by jrobinso** [comment]
For anyone still following this issue the sample names implementation is more or less complete in master and will be released in the next few weeks.   When loading a track that includes a concept of "samples", such as VCF with genotypes or "seg" copy number,  a new button will appear in the navbar,  "Samples",  clicking it will open a panel on the right with rows labeled by sample name.   This is also controllable in the "config" object.

**Comment by amanpatel** [comment]
Great news. I will check this out soon. Co-incidentally, I was working on the IGV integration in our app, and was glad to hear about this.

**Comment by jrobinso** [comment]
@amanpatel There are some developer options not documented yet.  I will do so here by example.  These are all optional and have defaults.

```
            showSampleNameButton:  true,
            showSampleNames: true,
            sampleNameViewportWidth: 200,
```

**Comment by jrobinso** [comment]
This is released now with 2.8.0
